### PR TITLE
[GPT-3] Support Grad Merge with FP32 main grad for BF16 training of GPT-3 model

### PR DESCRIPTION
**Pull request by haohongxiang**
[GPT-3] Support Grad Merge with FP32 main grad for BF16 training of GPT-3 model

### CODE REVIEW

File: ppfleetx/distributed/apis/amp.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+            assert param._grad_ivar() is None
+            if param.main_grad is not None:
+                assert param.main_grad.dtype == core.VarDesc.VarType.FP32
+                param_grads.append(param.main_grad)
+
+    temp_found_inf = to_variable(np.array([0]).astype(np.bool_))
+    if len(param_grads):
+        _legacy_C_ops.check_finite_and_unscale(
+            param_grads,
+            self._scale,
```
Comment (aoyulong) [inline ppfleetx/distributed/apis/amp.py:214]: What does the `self` refer to?

File: ppfleetx/optims/optimizer.py [hunk 2]
```diff
@@ -43,7 +43,7 @@ def __init__(self, learning_rate, parameters, grad_clip, **config):
         else:
             decay_params = [
                 p.name for p in parameters
-                if not any(nd in p.name for nd in ["bias", "norm"])
+                if not any(nd in p.name for nd in ["bias", "norm", "b_0"])
```
Comment (aoyulong) [inline ppfleetx/optims/optimizer.py:46]: Are these names fixed in different Paddle version?

File: ppfleetx/core/engine/eager_engine.py [hunk 3]
```diff
@@ -176,15 +177,21 @@ def configure_optimizers(self):
...
                     init_loss_scaling=self._scale_loss)
+            else:  # bfloat16
+                self._scaler = paddle.amp.GradScaler(
+                    init_loss_scaling=1, use_dynamic_loss_scaling=False)
 
             # Save dtype is the same as model dtype. Also can set save_dtype='float32' when
             # training with pure fp16 strategy, but will cause the rise of memory.
-            self._module.model = paddle.amp.decorate(
-                models=self._module.model, level='O2')
+            if self._amp_level == "O2":
```
Comment (aoyulong) [inline ppfleetx/core/engine/eager_engine.py:195]: Does the setting O2 and bfloat16 turn on the training of bfloat16? what's the meaning of the setting O1 and bfloat16?

**Review by ForFishes — APPROVED** [review]
LGTM

**Pull request merged**